pkg/cloud/services/eks/roles.go

```diff
@@ -203,7 +228,12 @@ func (s *NodegroupService) reconcileNodegroupIAMRole() error {
 		return errors.Wrapf(err, "error ensuring tags and policy document are set on node role")
 	}
 
-	policies := NodegroupRolePolicies()
+	var policies []string
+	if strings.Contains(s.scope.ControlPlane.Spec.Region, v1beta1.DefaultPartitionNameUSGov) {
```

Comment:
Instead of determining this partition all the time using if else we should do it one time and store the result 

and call functions with partition

Reply:
Currently 

> > /lgtm
> > @sadysnaat Are you happy with this too? :)
> 
> I think we need to refactor the duplicate code and avoid deducing partition using if else all the time
> 
> we should do this once and store in service.

Pls take look at the latest code, have made code changes to have util func to get a portion from the region, do we want to update service/crd to store partition ??